src/lib.rs

```diff
@@ -50,25 +50,23 @@ impl Preprocessor for D2 {
                 cmark(events, &mut buf).unwrap();
```

Comment:
the 'process_events' method is return a Result<impl Iterator, Error>, but the `cmark` method is expecting an `impl Iterator` (not wrapped in Result). You need to get the value from inside the `Result` by either handling or unwrapping the potential error value.
You can do this with the 'try' operator ('?') if the error type can be converted to the error type of the 'run' method. If not, you'll have to handle it some other way. I won't tell anyone if you want to just explicitly 'unwrap' it for now while you get the rest working and then come back and revisit it later

Reply:
Ah so I can't use ? here because it's in a closure that returns `()`